cdap-api/src/main/java/co/cask/cdap/api/workflow/AbstractWorkflow.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.workflow;
+
+import java.util.Map;
+
+/**
+ * Provides a default implementation of {@link Workflow} methods for easy extension.
+ */
+public abstract class AbstractWorkflow implements Workflow {
+
+  private WorkflowConfigurer configurer;
+
+  @Override
+  public final void configure(WorkflowConfigurer configurer) {
+    this.configurer = configurer;
+    configure();
+  }
+
+  /**
+   * Override this method to configure this {@link Workflow} job.
```

Comment:
Workflow Program or Job ?
@awholegunch : What do we usually call them in docs ?